File: PSet1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void){
    float input = 0;
    int cent = 0;
    int quarter = 0;
    int dime = 0;
    int nickel = 0;
    int remainder = 0;
    int coins = 0;
    
    do 
    {
        printf("How much change is owed?\n ");
        input = get_float();
        if(input == 0 || input <= 0){
            printf("Input should be greater than 0\n");
        }
    }
    while(input <= 0);
    
    cent = (int)round(input*100);

    quarter = cent / 25;
    remainder = cent % 25;
    
    dime = remainder / 10;
    remainder = remainder % 10;
    
    nickel = remainder / 5;
    remainder = remainder % 5;
    
    coins = quarter + dime + nickel + remainder;
    
    printf("Number of coins: %d\n", coins);

}

File: PSet6/greedy.py
import cs50
import math
import sys

while True:
    print('Change: ', end="")
    change = float(input())
    if(change == 0 or change <0):
        print('Input should be greater than 0')
    elif(change > 0):
        
        cent = math.floor(change * 100)
        
        quarter = int(cent / 25)
        remainder = int(cent % 25)
        
        dime = int(remainder / 10)
        remainder = int(remainder % 10)
        
        nickel = int(remainder / 5)
        remainder = int(remainder % 5)
        
        coins = int(quarter + dime + nickel + remainder)
        
        print(math.floor(coins))
        break
    else:
        print('Change: ', end="")
        change = float(input())